### PR TITLE
Make Steam client calls into a plugin

### PR DESCRIPTION
**Pull request by Atlante45**

### CODE REVIEW

File: interface/src/main.cpp [hunk 1]
```diff
@@ -129,8 +129,10 @@ int main(int argc, const char* argv[]) {
     }
 
     QCommandLineParser parser;
+    QCommandLineOption noSteamworks("noSteamworks", "Steamworks DLL not present");
```
Comment (birarda) [inline interface/src/main.cpp]: 😡

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5ac609497bc56ce43aacb7ecf888ad64840d16f3 available at: https://gist.github.com/176a472fb6568c15891ca668ac830132<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-5ac6094.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-5ac6094.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 98cb5a5fdb1b7c6edd9582907aa004215adad60d available at: https://gist.github.com/176a472fb6568c15891ca668ac830132<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-98cb5a5.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-98cb5a5.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4dc1a7d983d4dc56870a60342e4353eed5c5057e available at: https://gist.github.com/176a472fb6568c15891ca668ac830132<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-4dc1a7d.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-4dc1a7d.dmg

File: interface/src/main.cpp [hunk 2]
```diff
@@ -129,8 +128,10 @@ int main(int argc, const char* argv[]) {
     }
 
     QCommandLineParser parser;
+    QCommandLineOption checkMinSpecOption("checkMinSpec", "Check machine has minimum specifications");
```
Comment (birarda) [inline interface/src/main.cpp]: 😡

Comment (birarda) [inline interface/src/main.cpp]: Check if machine meets minimum specifications

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0dec97fc03b4e5cd393772cb70adb4b9de6bf70c available at: https://gist.github.com/176a472fb6568c15891ca668ac830132<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-0dec97f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-0dec97f.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bfc789f072f3e4f80a0abc5c06effc9169272541 available at: https://gist.github.com/176a472fb6568c15891ca668ac830132<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-bfc789f.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-bfc789f.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 588fdd6efab90856509dfcb5386f716746268f94 available at: https://gist.github.com/176a472fb6568c15891ca668ac830132<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-588fdd6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9211-588fdd6.dmg

**Comment by birarda** [comment]
QA'ed launching without `steam_api64.dll` and without `steamClient.dll`.

Smoke test passed. Further testing of functions that rely on Steam and the min spec guard are required.

**Pull request merged**